generate_makefile.bash

```diff
@@ -301,6 +303,18 @@ else
     KOKKOS_DEBUG_CMD=-DCMAKE_BUILD_TYPE=DEBUG
 fi
 
+if [ "$KOKKOS_HWLOC" == "" ]; then
```

Comment:
That's a bit silly.  If I change the definition above `ON -> OFF` and don't pay attention I will be surprised.
(Not requesting changes, just a comment.)

Reply:
Yeah, not sure why I added it this way, tired at the end of the day I guess. I'll change it.

Reply:
Change ot to checking for ON and make the empty declaration in the else clause.